diff --git a/setup.py b/setup.py
index 8677944..79d32d2 100644
--- a/setup.py
+++ b/setup.py
@@ -100,8 +100,8 @@ setup(
         "libstempo.spharmORFbasis",
         "libstempo.eccUtils",
     ],
-    setup_requires=["cython>=0.22", "numpy<2.0"],
-    install_requires=["numpy>=1.15.0,<2.0", "scipy>=1.2.0", "matplotlib>=3.3.2", "ephem>=3.7.7.1"],
+    setup_requires=["cython>=0.22", "numpy"],
+    install_requires=["numpy>=1.15.0", "scipy>=1.2.0", "matplotlib>=3.3.2", "ephem>=3.7.7.1"],
     extras_require={"astropy": ["astropy>=4.1"]},
     python_requires=">=3.7",
     ext_modules=cythonize(

