diff --git a/pyproject.toml b/pyproject.toml
index 37f283e..f364a8c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,9 +23,9 @@ exclude = '''

 [build-system]
 requires = [
-    "setuptools>=40.8.0",
+    "setuptools>=61",
     "wheel", # ephem package likes to have wheel installed
-    "Cython>=0.22",
-    "oldest-supported-numpy",
+    "cython>=3",
+    "numpy>=2.0.0rc1",
 ]
 build-backend = "setuptools.build_meta"
